services/workflows-service/src/transaction/transaction.repository.ts

```diff
@@ -111,14 +113,14 @@ export class TransactionRepository {
           subtractValue = getTransactionsParameters.timeValue * 24 * 60 * 60 * 1000;
           break;
         case TIME_UNITS.months:
-          now.setMonth(now.getMonth() - getTransactionsParameters.timeValue);
+          until.setMonth(until.getMonth() - getTransactionsParameters.timeValue);
           break;
         case TIME_UNITS.years:
-          now.setFullYear(now.getFullYear() - getTransactionsParameters.timeValue);
+          until.setFullYear(until.getFullYear() - getTransactionsParameters.timeValue);
           break;
       }
 
-      const pastDate = new Date(now.getTime() - subtractValue);
+      const pastDate = new Date(until.getTime() - subtractValue);
 
       whereClause.transactionDate = { gte: pastDate };
```

Comment:
Tricky issue here is that DateTime has time indication which not use only DATE filter 
So reset Time to 00:00:00 -> `until.setHours(0, 0, 0, 0)`